packages/notification-center/src/store/notifications-provider.context.tsx

```diff
@@ -1,195 +1,129 @@
-    }
-
-    if (!page.has(storeId)) {
-      setPage(page.set(storeId, 0));
-    }
-
-    if (isRefetch) {
-      notifications[storeId] = newNotifications;
-      setNotifications(Object.assign({}, notifications));
-    } else {
-      notifications[storeId] = [...(notifications[storeId] || []), ...newNotifications];
-      setNotifications(Object.assign({}, notifications));
-    }
-
-    setFetching(false);
-  }
-
-  async function fetchNextPage(storeId = 'default_store') {
-    if (!hasNextPage[storeId]) return;
-
-    const nextPage = page.get(storeId) + 1;
-    setPage(page.set(storeId, nextPage));
-
-    await fetchPage(nextPage, false, storeId);
-  }
-
-  async function markAsRead(messageId: string, storeId = 'default_store'): Promise<IMessage> {
-    notifications[storeId] = notifications[storeId].map((message) => {
-      if (message._id === messageId) {
-        message.read = true;
-        message.seen = true;
-      }
-
-      return message;
-    });
-
-    setNotifications(Object.assign({}, notifications));
-
-    return await api.markMessageAs(messageId, { seen: true, read: true });
-  }
+import React, { useState, useCallback, useMemo } from 'react';
+import type { IStoreQuery } from '@novu/client';
+import type { IMessage } from '@novu/shared';
 
-  async function markAllAsRead(storeId = 'default_store'): Promise<number> {
-    notifications[storeId] = notifications[storeId].map((message) => {
-      message.read = true;
-      message.seen = true;
-
-      return message;
-    });
-
-    setNotifications(Object.assign({}, notifications));
-
-    const messageIds = notifications[storeId].map((message) => {
-      return message._id;
-    });
-
-    return await api.markMessageAs(messageIds, { seen: true, read: true });
-  }
-
-  async function updateAction(
-    messageId: string,
-    actionButtonType: ButtonTypeEnum,
-    status: MessageActionStatusEnum,
-    payload?: Record<string, unknown>
-  ) {
-    await api.updateAction(messageId, actionButtonType, status, payload);
+import { NotificationsContext } from './notifications.context';
+import type { IStore } from '../shared/interfaces';
+import { useFetchNotifications, useUnseenCount } from '../hooks';
+import { useMarkNotificationsAs } from '../hooks/use-mark-notification-as.hook';
+
+const DEFAULT_STORES = [{ storeId: 'default_store' }];
+
+export function NotificationsProvider({
+  children,
+  stores = DEFAULT_STORES,
+}: {
+  children: React.ReactNode;
+  stores?: IStore[];
+}) {
+  const firstStore = stores[0];
+  const [storeQuery, setStoreQuery] = useState<IStoreQuery>(() => firstStore.query ?? {});
+  const [storeId, setStoreId] = useState(firstStore.storeId ?? 'default_store');
+  const setStore = useCallback(
+    (newStoreId: string) => {
+      const foundQuery = stores?.find((store) => store.storeId === newStoreId)?.query || {};
+      setStoreId(newStoreId);
+      setStoreQuery(foundQuery);
+    },
+    [stores, setStoreId, setStoreQuery]
+  );
```

Comment:
The `setStore` is called when the feed tab changes, we should just update the `storeId and query`